app/models/imprint/adapter.rb

```diff
@@ -0,0 +1,18 @@
+require 'dry-container'
+require 'dry-auto_inject'
+
+module Imprint
+  class Adapter
+    include Imprint::Wrapper.inject[:imprint_api]
+
+    def print_single_file(template_code, template_substitution, convert_to_pdf = false)
```

Comment:
Are ```template_code``` and ```template_substitution``` hashes? If so, better pass them as ```String``` and wrap in hash in building POST-request.